examples/cpp/flash_attn/flash_attn.cu

```diff
@@ -0,0 +1,376 @@
+    using BlockAdd = typename Config::BlockAdd;
+
+    using VecMax = typename Config::VecMax;
+    using VecAdd = typename Config::VecAdd;
+    using VecSub = typename Config::VecSub;
+    using VecMul = typename Config::VecMul;
+    using VecExp = typename Config::VecExp;
+
+    int block_x = CeilDiv<kM, kTM>;
+    int block_y = CeilDiv<kP, kTP>;
+    int block_z = kBatch;
+
+    dim3 grid(block_x, block_y, block_z);
+    dim3 block(Config::kThreads, 1, 1);
+
+    int shm_input = (kTM * kTK + kTK * kTN + kTN * kTP);
+    int shm_output = kTM * kTP;
+    int shm_size = shm_input < shm_output ? shm_output * sizeof(InType)
+                                          : shm_input * sizeof(InType);
+
+    auto kernel =
+        &KeFlashAttention<InType, AccType,
+                          OutType,                    //
+                          GIteratorA, SharedA, RegA,  //
+                          SharedALoader, RegALoader,  //
+                          GIteratorB, SharedB, RegB,  //
+                          SharedBLoader, RegBLoader,  //
+                          GIteratorC, SharedC, RegC,  //
+                          SharedCLoader, RegCLoader,  //
+                          RegAcc, RegAccCast, typename Config::GlobalD, RegD,
+                          RegDCast, DStorer, ConvertAcc, ConvertO, RegVec,
+                          CopyVec, RowMax, BroadcastSub, BroadcastMul,
+                          BroadcastDiv, BlockExp, BlockAdd, VecMax, VecAdd,
+                          VecSub, VecMul, VecExp>;
+
+    if (shm_size > 48 * 1024) {
+        cudaFuncSetAttribute(
+            kernel, cudaFuncAttributeMaxDynamicSharedMemorySize, shm_size);
+    }
+
+    kernel<<<grid, block, shm_size, 0>>>(A, B, C, D, kM, kN, kK, kP, kTM, kTN,
+                                         kTK, kTP);
+
+    // for (int i = 0; i < 5; ++i)
+    //     kernel<<<grid, block, shm_size, 0>>>(A, B, C, D, kM, kN, kK, kP, kTM,
+    //                                          kTN, kTK, kTP);
+    // cudaDeviceSynchronize();
+
+    // CudaTimer timer;
+
+    // int iter = 100;
+    // timer.start();
+    // for (int i = 0; i < iter; ++i)
+    //     kernel<<<grid, block, shm_size, 0>>>(A, B, C, D, kM, kN, kK, kP, kTM,
+    //                                          kTN, kTK, kTP);
+    // cudaDeviceSynchronize();
+
+    // float elapsed = timer.stop() / iter;
+    // std::cout << "Time: " << elapsed << " ms" << std::endl;
+}
+
+int main() {
+    run<FlashAttentionShape<64 /*M*/, 64 /*N*/, 128 /*K*/, 128 /*P*/>,
+        FlashAttentionShape<64 /*kTM*/, 64 /*kTN*/, 128 /*kTK*/, 128 /*kTP*/>,
+        1>();
+
+    // run<FlashAttentionShape<64 /*M*/, 64 /*N*/, 128 /*K*/, 128 /*P*/>,
+    //     FlashAttentionShape<64 /*kTM*/, 64 /*kTN*/, 128 /*kTK*/, 128
+    //     /*kTP*/>, 2>();
+
+    // run<B2BGemmShape<1024 /*M*/, 128 /*N*/, 128 /*K*/, 128 /*P*/>,
+    //     B2BGemmShape<64 /*kTM*/, 128 /*kTN*/, 128 /*kTK*/, 128 /*kTP*/>,
+    //     2>();
+
+    return 0;
```

Comment:
I think it is OK to delete line 344 ~ 359 and line 367~373. They can be added back when necessary in future.

Reply:
Deleted.